surveys/2024-annual-survey/questions.md

```diff
@@ -290,6 +290,23 @@ Type: select all that apply (optional)
 > in a separate question, though this is likely easier to find out through download
 > numbers.
 
+### What is your opinion on how fast Rust evolves?
+
+We want to know if you are able to keep up with the speed of development of the Rust language.
+
+Type: select one (optional)
+
+- Rust is already too complex, it should not add or stabilize any more features.
+- Rust changes too quickly, I wish it would slow down the pace of development.
+- I am satisfied with the current pace of development.
+- Rust changes too slowly, I wish it would add or stabilize features faster.
+- I don't know or don't care.
```

Comment:
What is the problem you're actually trying to solve here? Why were the previous options "bad"? Do you want more nuance? Or you just wanted a better question wording? I am not fully sold on these new set of options, I feel they are too close to the extreme boundaries of the spectrum of what a user can think.

- "Rust is already too complex, it should not add or stabilize any more features".
  Comment: seems to imply that Rust should just ... not evolve anymore? Like a legacy programming language (i.e. with existing codebases but no new code being written)?

- "Rust changes too slowly, I wish it would add or stabilize features faster".
  Comment: "_Pull requests welcome_"?

To my eyes, these options seem to invite people to express entitled opinions about the development of Rust, almost like in a employer-employee work relation. Are we working hard enough for you? :laughing:

Reply:
So, first let me say that the question that we had last year just.. sucked; I think that was kind of universally agreed upon, both from feedback on Reddit and also on [Zulip](https://rust-lang.zulipchat.com/#narrow/stream/402479-t-community.2Frust-survey/topic/2023.20Survey.20Feedback/near/408645264).

The original motivation for the question, as suggested by Jyn, was to get a vibe check about the speed with which Rust evolves. However, I worded the question very poorly:
```
I think that unimplemented features should be stabilised
vs
I wish the Rust project to not add major new features (or slow down the pace of development)
```

We asked if the user thinks that unimplemented features should be stabilised, which is just wrong for several reasons (is unimplemented = unstable, or just missing at all? why shouldn't they get stabilized? this probably depends on each specific feature? etc.). And the answers also presented a false dichotomy. And they missed a "I don't know" option.

So my first intention is to just nuke the question, as it was just bad.

The second intention is to replace it with a better question. I agree that the answers contain also the extreme sides of the spectrum, that was deliberate. I find it a bit baffling, but from what I've seen online, there are indeed voices that claim that Rust should just stop adding features, period, and just focus on polishing stuff. I find this view a bit too extreme, but it's certainly a valid opinion. Another valid opinion/request is people wanting Rust to evolve faster. They cannot really make any demands towards that, and as you said, if they want to help, they should start contributing, but that's true universally.

In the survey, we ask what is the users opinion about Rust; we don't promise anything. On the other hand, I think that it is quite interesting how do people see Rust's "change cadence". It probably won't happen, but if for some reason 95% people told us that they don't want tnew features, then as weird as it may sound, I think that it should be taken as an input for further development of Rust (again, I don't think that this will happen).

Reply:
Hmm ... I understand your overall reasoning but at the same time I can't parse it quite well :slightly_smiling_face: 

So, what are we looking for: feedback about the pace of development, right? I think one fundamental data point we gathered in 2023 survey was that the **No. 1 voted option** for "what are your biggest worries about Rust" was "becomes too complex". The way I see it, people are literaly screaming at us saying "Rust is a tough nut to crack".

This new question looks almost like a duplicate or a second, expanded version of "what are your biggest worries about Rust". What we _could_ ask is instead if how people feel about us working more on making Rust "easier" (blanket word for multiple concepts) or add "new stuff". The problem is that they intersect. To make things "easier" we sometimes need to add "new stuff".

Reply:
You're right on both accounts :) This is essentially a questions that "zooms-in" on the "becomes too complex" opinions. I don't see that as a bad thing, but I also don't have a strong opinion about it, I'm fine with just removing this question.

Regarding the second point: indeed there are some features that are more about "filling in gaps in the existing system" rather than "adding new features". For example, async fn in traits is essentially enabling something that people intuitively expected to work before. That's quite different from e.g. adding an effect system that could transform how Rust is written  in a large way.

However, as usually with these questions, the devil is in the details; it's hard to convey all these little nuances in a few words.. :)

Reply:
> if for some reason 95% people told us that they don't want new features (...) (I don't think that this will happen)

if _that_ happens I'm outta here LOL